packages/components/icon/src/icon.ts

```diff
@@ -39,40 +43,47 @@ export class Icon extends LitElement {
   /**
    * Add icon(s) to the icon registry
    *
-   * @param {IconDefinition | IconDefinition[] } faIcons One or more IconDefinition that have been imported from FontAwesome
+   * @param {IconDefinition | IconDefinition[] } icons One or more IconDefinition that have been imported from FontAwesome
    */
-  static registerIcon(...faIcons: IconDefinition[]): void {
-    // TODO: find a better (and more universal) way to only log these kind of warnings in dev mode
-    const isDevMode = location.hostname === 'localhost';
-
-    faIcons.forEach(icon => {
-      if (window.SLDS.icons[`${icon.prefix}-${icon.iconName}`] && isDevMode) {
-        console.warn(`Icon ${icon.prefix}-${icon.iconName} is already in the registry`);
-        return;
-      }
-
-      const {
-          icon: [width, height, , , path]
-        } = icon,
-        paths = Array.isArray(path) ? path : [path];
-      const svg = `
-        <svg viewBox="0 0 ${width} ${height}" "xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
-          ${paths
-            .map((p: string, i) => `<path d="${p}" fill="var(--fill-${Icon.getColorToken(i, icon.prefix)})"></path>`)
-            .join('')}
-        </svg>
-      `;
-
-      window.SLDS.icons[`${icon.prefix}-${icon.iconName}`] = { svg, type: 'RegisteredIcon' };
-    });
-  }
+  static register(...icons: IconDefinition[]): void;
 
   /**
    * Store all icons from the IconLibrary of the theme (icons.json) in the icon registry for easy access.
    * Is run in the setup method of each theme.
+   *
+   * @param {IconLibrary} icons The IconLibrary of the theme
    */
-  static registerIcons(icons: IconLibrary): void {
-    window.SLDS.icons = { ...window.SLDS.icons, ...icons };
+  static register(icons: IconLibrary): void;
+
+  static register(icon: IconDefinition | IconLibrary, ...icons: IconDefinition[]): void {
+    // TODO: find a better (and more universal) way to only log these kind of warnings in dev mode
```

Comment:
So lit does this: `if (process.env.NODE_ENV === 'development') { console.warn(....) }` which I've seen elsewhere as well. But I'm not sure about the implications. Lit also *ships* separate development and production bundles. And those are specified in `package.json`. Something to think about.